### PR TITLE
Only encode data with <128 bytes in base58

### LINKED ISSUE #18929: accountSubscribe with encoding:base58 may lead to validator lags
#### Problem
_Steps_
1. Take account that has large data field e.g. `58KcficuUqPDcMittSddhT8LzsPJoH46YP4uURoMo5EB`
2. Send 
`{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "accountSubscribe",
  "params": [
    "58KcficuUqPDcMittSddhT8LzsPJoH46YP4uURoMo5EB",
    {
      "encoding": "base58",
      "commitment": "finalized"
    }
  ]
}`
3. Wait some time (you may optionally monitor using base64/base64+zstd subscriptions, until several notifications )

_Observed_
Validator hangs and does not accept new WS connections

#### Explanation
Account `58KcficuUqPDcMittSddhT8LzsPJoH46YP4uURoMo5EB` has over 1Mb data. 
`Base58` encoder has quadratic complexity, therefore it stuck with such input for some minutes which leads to denial of service of whole notification process
https://github.com/mycorrhiza/bs58-rs/blob/pr%C4%ABmum/src/encode.rs#L335

#### Proposed Solution
As it does not look feasible to verify data field size when validator receives subscription request, we propose 2 possible solutions
1. Check account data field size at notification moment and return some flag, reflecting error
2. Do not allow to use `base58` in subscriptions or forcing it to base64 (as done with jsonParsed encoding for some accounts)

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

base58 encoding is slow

#### Summary of Changes

Only encode with account data 128 bytes or less. This means that account and program subscriptions that would return large base58-encoded account data will now return an error string for that field.

Fixes #18929

### CODE REVIEW

File: account-decoder/src/lib.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ use {
 
 pub type StringAmount = String;
 pub type StringDecimals = String;
+pub const MAX_BS58_BYTES: usize = 128;
```
Comment (mvines) [inline account-decoder/src/lib.rs:31]: Help a brother out:
```suggestion
pub const MAX_BASE58_BYTES: usize = 128;
```
BS means something else to most people

File: account-decoder/src/lib.rs [hunk 2]
```diff
@@ -60,6 +61,17 @@ pub enum UiAccountEncoding {
...
+        account: &T,
+        data_slice_config: Option<UiDataSliceConfig>,
+    ) -> String {
+        if account.data().len() <= MAX_BS58_BYTES {
+            bs58::encode(slice_data(account.data(), data_slice_config)).into_string()
+        } else {
+            "error: data too large for bs58 encoding".to_string()
+        }
+    }
+
```
Comment (mvines) [inline account-decoder/src/lib.rs:74]:
```suggestion
    fn encode_base58<T: ReadableAccount>(
        account: &T,
        data_slice_config: Option<UiDataSliceConfig>,
    ) -> String {
        if account.data().len() <= MAX_BASE58_BYTES {
            bs58::encode(slice_data(account.data(), data_slice_config)).into_string()
        } else {
            "error: data too large for base58 encoding".to_string()
        }
    }

```

File: account-decoder/src/lib.rs (hunk 1 above)
Comment (t-nelson) [reply account-decoder/src/lib.rs]: I dunno, still seems applicable

File: account-decoder/src/lib.rs (hunk 2 above)
Comment (CriesofCarrots) [reply account-decoder/src/lib.rs]: https://github.com/solana-labs/solana/pull/19317/commits/66fa63b86c6d72400a66446c84a82a678e2cd72b

**Comment by CriesofCarrots** [comment]
So far, data suggests downstream breakage will be small to none. Holding off on merge for more data.

**Comment by CriesofCarrots** [comment]
No usage of base58 encoding for account or program subscriptions evidenced on our pools. Merging!

**Pull request merged**